src/app/clusters/time-synchronization-server/time-synchronization-delegate.h

```diff
@@ -104,12 +104,17 @@ class Delegate
      * a CHIP_ERROR.
      */
     virtual CHIP_ERROR UpdateTimeUsingNTPFallback(const CharSpan & fallbackNTP,
-                                                  chip::Callback::Callback<OnFallbackNTPCompletion> * callback) = 0;
+                                                  chip::Callback::Callback<OnFallbackNTPCompletion> * callback)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
 
     /**
      * @brief Signals application that UTCTime has changed through the timesync cluster.
      */
-    virtual void UTCTimeAvailabilityChanged(uint64_t time) = 0;
+    virtual void UTCTimeAvailabilityChanged(uint64_t time) {}
+    virtual void TrustedTimeSourceAvailabilityChanged(bool available, GranularityEnum granularity) {}
```

Comment:
What's the use case for this?

Reply:
This is for the apps to be able to manage their time origin and decide if it needs to query for time.
Right now when TrustedTimeSource is set, the code attempt to get time from the source only if the current granularity is none. But this might be irrelevant for some apps.